src/ape/plugins/_utils.py

```diff
@@ -344,12 +344,17 @@ def __str__(self):
         version_key = f"=={self.version}" if self.version and self.version[0].isnumeric() else ""
         return f"{self.name}{version_key}"
 
-    def check_installed(self, use_cache: bool = True):
+    def check_installed(self, use_cache: bool = True) -> bool:
         if not use_cache:
             _get_distributions.cache_clear()
 
-        ape_packages = [n.name for n in _get_distributions()]
-        return self.package_name in ape_packages
+        for dist in _get_distributions():
+            if not (name := getattr(dist, "name", "")):
+                continue
+            elif self.package_name == name:
```

Comment:
Could we reduce this one leg?
```suggestion
            if self.package_name == getattr(dist, "name", ""):
```